### DISCUSSION TITLE
Airport security r simmer

### DISCUSSION THREAD

**Discussion by Jbbbb777** [r-simmer/simmer#290] [Q&A]
I am trying to do a security check in an airport and i am meant to use a poisson distribuition instead of a exponential. Also i have 22 counters which it has a singular queue or one queue for each counter. The plots are not great so i may have some errors.
Security time consists in preparation time plus scanning time.
It is supposed to be an 1 hour simulation.

This is the code

env <- simmer("airport")
env
lambda = 1384/60

passenger <- trajectory("passenger path") %>%
log_("Arrived") %>%
seize("validation machine", 1) %>%
timeout(function() runif(1,.016,.05 )) %>%
release("validation machine", 1) %>%
log_("Validated") %>%

seize("counter") %>%
timeout(function() rexp(1, lambda)) %>%
release("counter") %>%
log_("arrived security") %>%

seize("security time",1) %>%
timeout(function() rtri(1,min= .17 ,max= .75, mode= .33)) %>%
release("security time", 1) %>%
log_("Vou-me embora(security time)")

env %>%
add_resource("validation machine", 3) %>%
add_resource("counter", 22) %>%
add_resource("security time",1) %>%
add_generator("passenger", passenger, function() rexp(1 , lambda ))

env %>% run(until= 60)

**Comment by Enchufa2** [comment]
Note that timeouts and generator functions produce service times and interarrival times respectively. Therefore, I think it doesn't make sense to use a Poisson _distribution_ for any of them. If you meant that you want to use a **Poisson process**, then interarrivals are exponentially distributed, so the exponential distribution is the correct one to apply.

As for counters, with the setup above you have 22 counters with a single queue. For 22 independent queues, you can use instead `add_resource(paste0("counter", 1:22))`, then you need to add a `select(paste0("counter", 1:22))` in your trajectory, and finally `seize_selected()` and `release_selected()`.

**Reply by Jbbbb777** [reply]
Hey thanks, so i did that and it worked, also i do not think the timeout function on counter is right i just used the triangular because it is the preparation and scanning time. But for the one queue that i did is it right? since there is a mean of 1384 passengers per hour.
Also i wanted to add 4 new counters for priority passengers. Do i use a branch? i do not really know how to do it.
I get this feeling that it is not right since it arrives 23 passengers per minute to 22 counters, is it supposed to not have queues??
Thanks!!


The code regarding your help

env <- simmer("airport")
env 
lambda = 1384/60



passenger <- trajectory("passenger path") %>%
    log_("Arrived") %>% 
  seize("validation machine", 1) %>%
  timeout(function() runif(1,.016,.05 )) %>%
  release("validation machine", 1) %>%
    log_("Validated") %>%

  select(paste0("counter", 1:22)) %>% 
  seize_selected() %>% 
  timeout(function() rtri(1,min= .17 ,max= .75, mode= .33)) %>% 
  release_selected() %>% 
    log_("left security") 

env %>% 
  add_resource("validation machine", 3) %>%
  add_resource(paste0("counter", 1:22)) %>% 
  add_generator("passenger", passenger, function() rexp(1 , lambda ))
    
env %>% run(until= 60)

**Reply by Jbbbb777** [reply]
Hello can you please help me as soon as possible?? thank you for your help

**Reply by Enchufa2** [reply]
Yes, you can add more resources and redirect priority arrivals to those using a branch e.g. based on some probability. There are multiple examples of branches in the vignettes.

I'm not sure what you mean when you say it doesn't look right. The simulation looks ok to me.
